src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -0,0 +1,374 @@
+            }
+            else
+            {
+                Assert.Fail("Test summary not found");
+            }
+        }
+        catch (Exception e)
+        {
+            if (string.IsNullOrEmpty(output))
+            {
+                output = e.Message;
+            }
+            Assert.Fail($"Build failed: {output}");
+        }
+        finally
+        {
+            _dockerHelper.DeleteImage(tag);
+        }
+    }
+
+    private string GenerateDockerfile(List<string> rpmPackageList, string baseImage, DistroType distroType)
+    {
+        StringBuilder sb = new();
+        sb.AppendLine("FROM " + baseImage);
+        sb.AppendLine("");
+        sb.AppendLine("# Copy NuGet.config");
+        sb.AppendLine($"COPY NuGet.config .");
+
+        sb.AppendLine("");
+        sb.AppendLine("# Copy scenario-tests content");
+        sb.AppendLine($"COPY scenario-tests scenario-tests");
+
+        sb.AppendLine("");
+        sb.AppendLine("# Copy nuget packages");
+        sb.AppendLine($"COPY packages packages");
+
+        sb.AppendLine("");
+        sb.AppendLine("# Copy RPM packages");
+        foreach (string package in rpmPackageList)
+        {
+            sb.AppendLine($"COPY {package} {package}");
+        }
+        sb.AppendLine("");
+        sb.AppendLine("# Install RPM packages and Microsoft.DotNet.ScenarioTests.SdkTemplateTests tool");
+        sb.Append("RUN");
+
+        // TODO: remove --force-all when aspnet package versioning issue have been resolved - https://github.com/dotnet/source-build/issues/4895
+        string packageInstallationCommand = distroType == DistroType.Deb ? "dpkg -i --force-all" : "rpm -i";
+        bool useAndOperator = false;
+        foreach (string package in rpmPackageList)
+        {
+            sb.AppendLine(" \\");
+            sb.Append($"    {(useAndOperator ? "&&" : "")} {packageInstallationCommand} {package}");
+            useAndOperator = true;
+        }
+        sb.AppendLine("");
+
+        // Set environment for nuget.config
+        sb.AppendLine("");
+        sb.AppendLine("# Set custom nuget.config");
+        sb.AppendLine("ENV RestoreConfigFile=/NuGet.config");
+
+        // Find scenario-tests binary in context/scenario-tests
+        string? scenarioTestsBinary = Directory.GetFiles(Path.Combine(_contextDir, "scenario-tests"), "Microsoft.DotNet.ScenarioTests.SdkTemplateTests.dll", SearchOption.AllDirectories).FirstOrDefault();
+        if (scenarioTestsBinary == null)
+        {
+            throw new Exception("Scenario tests binary not found");
+        }
+        scenarioTestsBinary = scenarioTestsBinary.Replace(_contextDir, "").Replace("\\", "/");
+
+        // Set entry point
+        sb.AppendLine("");
+        sb.AppendLine($"ENTRYPOINT [ \"dotnet\", \"{scenarioTestsBinary}\", \"--dotnet-root\", \"/usr/share/dotnet\" ]");
+
+        string dockerfile = Path.Combine(_contextDir, Path.GetRandomFileName());
```

Comment:
I feel like debugging would be easier if the filename started with `Dockerfile` with a random suffix.

Reply:
Fixed with https://github.com/dotnet/sdk/pull/46927/commits/08a423071d13796dd1ac99a900ccae15ce1705c6